src/pkg/services/m365/api/pagers/pagers.go

```diff
@@ -214,7 +220,10 @@ func EnumerateItems[T any](
 		pager.SetNextLink(nextLink)
 	}
 
-	logger.Ctx(ctx).Infow("completed delta item enumeration", "result_count", len(result))
+	logger.Ctx(ctx).Infow(
+		"completed item enumeration",
+		"result_count", resultCount,
```

Comment:
nit: `itemCount` might be more descriptive.
```suggestion
		"item_count", itemCount,
```